code/datums/elements/_element.dm

```diff
@@ -52,10 +52,11 @@
 /datum/proc/_AddElement(list/arguments)
 	if(QDELING(src))
 		CRASH("We just tried to add an element to a qdeleted datum, something is fucked")
+	var/element_type = arguments[1]
```

Comment:
This has a cpu cost

Reply:
in case anyone is :eyes:ing this, AddElement is called 80,000 times during init and so if it's very simple to keep zero-cost (like it is here), then we would prefer that